frictionless/portals/github/manager.py

```diff
@@ -1,45 +1,156 @@
+import os
+from collections import namedtuple
 from typing import Optional
 from .control import GithubControl
-from ...platform import platform
-from ...resource import Resource
+from ...exception import FrictionlessException
 from ...package import Package
 from ...package import Manager
+from ...platform import platform
+from ...resource import Resource
 
 
 class GithubManager(Manager[GithubControl]):
 
     # Read
 
-    # TODO: implement
-    def read_catalog(self):
-        pass
+    def read_catalog(self, user: Optional[str] = None, apikey: Optional[str] = None):
+        client = platform.github.Github(apikey, per_page=self.control.per_page)
+        user = user or self.control.user
+        search = self.control.search or f"user:{user}"
+        print(search)
+        assert search
+        if self.control.page:
+            search = f"{search}&page={self.control.page}"
```

Comment:
Maybe using Python's builtin means for building query string. It's not really important but you can try searching for urlib.parse for query strings (can't remember the function)

Reply:
I have removed the query string generation part.